snapcraft/plugins/python.py

```diff
@@ -159,219 +143,173 @@ def plugin_stage_packages(self):
 
     @property
     def stage_packages(self):
-        if not os.path.exists(self._get_python_command()):
+        try:
+            _python.get_python_command(
+                self._python_major_version, stage_dir=self.project.stage_dir,
+                install_dir=self.installdir)
+        except _python.errors.MissingPythonCommandError:
             return super().stage_packages + self.plugin_stage_packages
         else:
             return super().stage_packages
 
+    @property
+    def _pip(self):
+        if not self.__pip:
+            self.__pip = _python.Pip(
+                python_major_version=self._python_major_version,
+                part_dir=self.partdir,
+                install_dir=self.installdir,
+                stage_dir=self.project.stage_dir)
+        return self.__pip
+
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
         self.build_packages.extend(self.plugin_build_packages)
-        self._python_package_dir = os.path.join(self.partdir, 'packages')
         self._manifest = collections.OrderedDict()
 
+        # Pip requires only the major version of python rather than the command
+        # name like our option requires.
+        match = re.match(
+            'python(?P<major_version>\d).*', self.options.python_version)
+        if not match:
+            raise UnsupportedPythonVersionError(
+                python_version=self.options.python_version)
+
+        self._python_major_version = match.group('major_version')
+        self.__pip = None
+
     def pull(self):
         super().pull()
 
-        setup = 'setup.py'
-        if os.listdir(self.sourcedir):
-            setup = os.path.join(self.sourcedir, 'setup.py')
+        self._pip.setup()
+
         with simple_env_bzr(os.path.join(self.installdir, 'bin')):
```

Comment:
this might not (certainly since the start of our use of sitecustomize) is not needed anymore.